### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by gixxerblade**
## Description
Created dark mode for the site. 

## Related Issue
#90 

## Motivation and Context
Makes UI more enjoyable

## How Has This Been Tested?
Visually and Jest. All tests passed

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by vodafone001 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -53,7 +54,7 @@
     "format:src": "prettier --write \"src/**/*.js\"",
     "format:templates": "prettier --write \"src/templates/**/*.js\"",
     "format:test": "prettier --write \"tests/**/*.js\"",
-    "gatsby:develop": "gatsby develop",
+    "gatsby:develop": "gatsby develop -H 0.0.0.0",
```
Comment (andrew-t-james) [inline package.json:57]: Please revert.

File: src/components/Header/Header.js [hunk 2]
```diff
@@ -2,7 +2,8 @@ import React from 'react'
 import { graphql, useStaticQuery, Link } from 'gatsby'
 import BackgroundImage from 'gatsby-background-image'
 import Typed from 'react-typed'
-
+import Toggle from '../Toggle'
+import '../../assets/css/custom.css'
```
Comment (andrew-t-james) [inline src/components/Header/Header.js:6]: Add space between imports and function.

File: src/components/Nav/Nav.js [hunk 3]
```diff
@@ -1,9 +1,8 @@
 import React, { useState, useRef, useEffect } from 'react'
 import Link from 'gatsby-link'
 import FluidImage from '../FluidImage'
-
 import './nav.css'
-
+import Toggle from '../Toggle'
```
Comment (andrew-t-james) [inline src/components/Nav/Nav.js:5]: Add space between imports and function move this import above the css import.

File: src/components/SponsorSlider/SponsorSlider.js [hunk 4]
```diff
@@ -17,81 +21,115 @@ const alignmentStyles = {
...
+  },
+  {
+    href: 'https://www.contentful.com/',
+    label: 'Link to Contentful',
+    title: 'Contentful',
+    lightElement: 'contentful.png',
+    darkElement: <Contentful size={`60`} color="#091f40" />,
+  },
+]
 function SponsorSlider() {
```
Comment (andrew-t-james) [inline src/components/SponsorSlider/SponsorSlider.js:68]: Add space between function and elements array.

File: src/components/SponsorSlider/SponsorSlider.js [hunk 5]
```diff
@@ -17,81 +21,115 @@ const alignmentStyles = {
...
-  const [viewport, setViewport] = useState(800)
+  const { colorMode } = useContext(ThemeContext)
 
+  const [viewport, setViewport] = useState(800)
+  //const [colorMode, setColorMode] = useState(undefined)
```
Comment (andrew-t-james) [inline src/components/SponsorSlider/SponsorSlider.js:72]: Remove comment.

File: src/components/SponsorSlider/SponsorSlider.js [hunk 6]
```diff
@@ -17,81 +21,115 @@ const alignmentStyles = {
...
   function updateWindowDimensions() {
     setViewport(window.innerWidth)
   }
 
   useEffect(() => {
     window.addEventListener('resize', updateWindowDimensions)
-
+    //const root = window.document.documentElement
```
Comment (andrew-t-james) [inline src/components/SponsorSlider/SponsorSlider.js:79]: Remove comments.

File: src/components/Toggle/Toggle.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        />
+        {colorMode === 'light' ? <FaMoon size={size} /> : <FaSun size={size} />}
+      </label>
+    </div>
+  )
+}
+Toggle.propTypes = {
+  size: PropTypes.number,
+}
+export default Toggle
```
Comment (andrew-t-james) [inline src/components/Toggle/Toggle.js:33]: Add line breaks between prop types and function, and below between imports.

File: src/icons/Contentful.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react'
+import PropTypes from 'prop-types'
+function SvgContentful({ size, color, style }) {
```
Comment (andrew-t-james) [inline src/icons/Contentful.js:3]: Same for this file and line breaks between imports, function, prop types, and export.

File: src/icons/Contentful.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react'
```
Comment (andrew-t-james) [inline src/icons/Contentful.js:1]: Please change to import React from 'react'

**Review by andrew-t-james — CHANGES_REQUESTED** [review]

File: src/components/Header/Header.js [hunk 10]
```diff
@@ -2,6 +2,8 @@ import React from 'react'
 import { graphql, useStaticQuery, Link } from 'gatsby'
 import BackgroundImage from 'gatsby-background-image'
 import Typed from 'react-typed'
+import Toggle from '../Toggle'
+import '../../assets/css/custom.css'
```
Comment (andrew-t-james) [inline src/components/Header/Header.js:6]: Please remove this import.

File: src/icons/Fem.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react'
+import PropTypes from 'prop-types'
+function SvgFem({ size, color, ...style }) {
```
Comment (andrew-t-james) [inline src/icons/Fem.js:3]: Please and line breaks to this file as well.

File: package.json [hunk 12]
```diff
@@ -4,6 +4,7 @@
   "repository": "Vets-Who-Code/vwc-site",
   "dependencies": {
     "@hot-loader/react-dom": "16.13.0",
+    "@svgr/cli": "^5.4.0",
```
Comment (andrew-t-james) [inline package.json:7]: please remove this unused package.

File: assets/fem.svg [hunk 13]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (andrew-t-james) [inline assets/fem.svg:1]: Please remove unused svg files from assets.

**Review by andrew-t-james — CHANGES_REQUESTED** [review]

File: src/components/Header/Header.js (hunk 10 above)
Comment (gixxerblade) [reply src/components/Header/Header.js:6]: Done

File: src/icons/Fem.js (hunk 11 above)
Comment (gixxerblade) [reply src/icons/Fem.js:3]: Done. Also import React from 'react'

File: assets/fem.svg (hunk 13 above)
Comment (gixxerblade) [reply assets/fem.svg:1]: Removed

**Review by andrew-t-james — APPROVED** [review]

**Pull request merged**